browser/importer/brave_profile_writer.cc

```diff
@@ -1,519 +0,0 @@
-    msg << "An error occurred getting wallet properties "
-      << "(error_code=" << error_code << ")";
-    CancelWalletImport(msg.str());
-    return;
-  }
-}
-
-void BraveProfileWriter::OnRecoverWallet(
-    brave_rewards::RewardsService* rewards_service,
-    unsigned int result,
-    double balance) {
-  rewards_service->RemoveObserver(this);
-
-  if (result) {
-    // Cancel the import if wallet restore failed
-    std::ostringstream msg;
-    msg << "An error occurred while trying to restore "
-      << "the wallet (result=" << result << ")";
-    CancelWalletImport(msg.str());
-    return;
-  }
-
-  LOG(INFO) << "Wallet restore successful";
-  SetWalletProperties(rewards_service);
-
-  // Set the pinned item count (rewards can detect and take action)
-  PrefService* prefs = profile_->GetPrefs();
-  prefs->SetInteger(kBravePaymentsPinnedItemCount, pinned_item_count_);
-
-  // Notify the caller that import is complete
-  DCHECK(bridge_ptr_);
-  bridge_ptr_->FinishLedgerImport();
-}
-
-void BraveProfileWriter::CancelWalletImport(std::string msg) {
-  if (IsInObserverList()) {
-    rewards_service_->RemoveObserver(this);
-  }
-  LOG(ERROR) << msg;
-  DCHECK(bridge_ptr_);
-  // NOTE: calling bridge_ptr_->Cancel() may roll back previously imported
-  // items. Instead, let's let the importer finish (errors are logged).
-  bridge_ptr_->FinishLedgerImport();
-}
-
-void BraveProfileWriter::SetWalletProperties(brave_rewards::RewardsService*
-  rewards_service) {
-  // Set the preferences read from session-store-1
-  auto* payments = &ledger_.settings.payments;
-  rewards_service->SetPublisherAllowVideos(payments->allow_media_publishers);
-  rewards_service->SetPublisherAllowNonVerified(payments->allow_non_verified);
-  rewards_service->SetPublisherMinVisitTime(payments->min_visit_time);
-  rewards_service->SetPublisherMinVisits(payments->min_visits);
-
-  // Set the excluded sites
-  for (const auto& publisher_key : ledger_.excluded_publishers) {
-    rewards_service->SetPublisherExclude(publisher_key, true);
-  }
-
-  // Set the recurring tips (formerly known as pinned sites)
-  int sum_of_monthly_tips = 0;
-  pinned_item_count_ = 0;
-  for (const auto& publisher : ledger_.pinned_publishers) {
-    // NOTE: this will truncate (ex: 0.90 would be 0, not 1)
-    const int amount_in_bat =
-        static_cast<int>((publisher.pin_percentage / 100.0) *
-                         ledger_.settings.payments.contribution_amount);
-    if (amount_in_bat > 0) {
-      pinned_item_count_++;
-      sum_of_monthly_tips += amount_in_bat;
-
-      // Add publisher to `publisher_info`
-      auto site = std::make_unique<brave_rewards::ContentSite>();
-      site->id = publisher.key;
-      site->status = 0;
-      site->excluded = 0;
-      site->name = publisher.name;
-      site->favicon_url = "";
-      site->url = publisher.url;
-      site->provider = publisher.provider;
-
-      // Add `recurring_donation` entry
-      rewards_service->OnTip(publisher.key,
```

Comment:
based on the fact that this one is deleted @simonhong could you please remove `OnTip` function from rewards_service that accepts this params as it was only kept for this use case. Thank you

Reply:
There are one more user of that api - https://github.com/brave/brave-core/blob/master/components/brave_rewards/browser/rewards_service_browsertest.cc#L1469

Reply:
@NejcZdovc I tried to cleanup test files also with below patch. but got many test failure.
I think this cleanup should be done as a f/u.

```
diff --git a/components/brave_rewards/browser/rewards_service.h b/components/brave_rewards/browser/rewards_service.h
index 14c36c1c0..5091f022b 100644
--- a/components/brave_rewards/browser/rewards_service.h
+++ b/components/brave_rewards/browser/rewards_service.h
@@ -206,13 +206,6 @@ class RewardsService : public KeyedService {
       const double amount,
       const bool recurring) = 0;
 
-  // Used in importer from muon days
-  virtual void OnTip(
-      const std::string& publisher_key,
-      double amount,
-      const bool recurring,
-      std::unique_ptr<brave_rewards::ContentSite> site) = 0;
-
   virtual void RemoveRecurringTipUI(const std::string& publisher_key) = 0;
   virtual void GetRecurringTipsUI(GetRecurringTipsCallback callback) = 0;
   virtual void GetOneTimeTipsUI(GetOneTimeTipsCallback callback) = 0;
diff --git a/components/brave_rewards/browser/rewards_service_browsertest.cc b/components/brave_rewards/browser/rewards_service_browsertest.cc
index effd11e7e..213b97550 100644
--- a/components/brave_rewards/browser/rewards_service_browsertest.cc
+++ b/components/brave_rewards/browser/rewards_service_browsertest.cc
@@ -1453,20 +1453,12 @@ class BraveRewardsBrowserTest
   void TipViaCode(
       const std::string publisher_key,
       const double amount,
-      const ledger::PublisherStatus status,
       const bool should_contribute = false,
       const bool recurring = false,
       const ledger::Result result = ledger::Result::LEDGER_OK) {
     tip_reconcile_completed_ = false;
     pending_tip_saved_ = false;
-    auto site = std::make_unique<brave_rewards::ContentSite>();
-    site->id = publisher_key;
-    site->name = publisher_key;
-    site->url = publisher_key;
-    site->status = static_cast<int>(status);
-    site->provider = "";
-    site->favicon_url = "";
-    rewards_service_->OnTip(publisher_key, amount, recurring, std::move(site));
+    rewards_service_->OnTip(publisher_key, amount, recurring);
 
     if (recurring) {
       return;
@@ -2388,14 +2380,12 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       20.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
   TipViaCode(
       "brave.com",
       50.0,
-      ledger::PublisherStatus::NOT_VERIFIED,
       false,
       true);
 
@@ -2426,14 +2416,12 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       50.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
   TipViaCode(
       "brave.com",
       50.0,
-      ledger::PublisherStatus::NOT_VERIFIED,
       false,
       true);
 
@@ -2476,12 +2464,12 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   EXPECT_TRUE(WaitForLoadStop(contents()));
 
   // Tip unverified publisher
-  TipViaCode("brave.com", 1.0, ledger::PublisherStatus::NOT_VERIFIED);
-  TipViaCode("brave.com", 5.0, ledger::PublisherStatus::NOT_VERIFIED);
-  TipViaCode("3zsistemi.si", 10.0, ledger::PublisherStatus::NOT_VERIFIED);
-  TipViaCode("3zsistemi.si", 5.0, ledger::PublisherStatus::NOT_VERIFIED);
-  TipViaCode("3zsistemi.si", 10.0, ledger::PublisherStatus::NOT_VERIFIED);
-  TipViaCode("3zsistemi.si", 10.0, ledger::PublisherStatus::NOT_VERIFIED);
+  TipViaCode("brave.com", 1.0);
+  TipViaCode("brave.com", 5.0);
+  TipViaCode("3zsistemi.si", 10.0);
+  TipViaCode("3zsistemi.si", 5.0);
+  TipViaCode("3zsistemi.si", 10.0);
+  TipViaCode("3zsistemi.si", 10.0);
 
   // Claim promotion using panel
   ClaimPromotion(true);
@@ -2568,7 +2556,6 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, RewardsPanelDefaultTipChoices) {
   TipViaCode(
       "3zsistemi.si",
       10.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
@@ -2679,7 +2666,6 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       amount,
-      ledger::PublisherStatus::VERIFIED,
       should_contribute);
   VerifyTip(amount, should_contribute, false, true);
 
@@ -2712,14 +2698,12 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       amount,
-      ledger::PublisherStatus::VERIFIED,
       should_contribute);
   total_amount += amount;
 
   TipViaCode(
       "laurenwags.github.io",
       amount,
-      ledger::PublisherStatus::VERIFIED,
       should_contribute);
   total_amount += amount;
 
@@ -2752,7 +2736,6 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, TipConnectedPublisherAnon) {
   TipViaCode(
       "bumpsmack.com",
       amount,
-      ledger::PublisherStatus::CONNECTED,
       should_contribute);
   VerifyTip(amount, should_contribute, false, true);
 }
@@ -2785,7 +2768,6 @@ IN_PROC_BROWSER_TEST_F(
   TipViaCode(
       "bumpsmack.com",
       amount,
-      ledger::PublisherStatus::CONNECTED,
       should_contribute);
   VerifyTip(amount, should_contribute, false, true);
 }
@@ -2816,7 +2798,6 @@ IN_PROC_BROWSER_TEST_F(
   TipViaCode(
       "bumpsmack.com",
       amount,
-      ledger::PublisherStatus::CONNECTED,
       should_contribute,
       false,
       ledger::Result::LEDGER_ERROR);
@@ -2850,7 +2831,6 @@ IN_PROC_BROWSER_TEST_F(
   TipViaCode(
       "bumpsmack.com",
       amount,
-      ledger::PublisherStatus::CONNECTED,
       should_contribute,
       false,
       ledger::Result::LEDGER_ERROR);
@@ -2908,7 +2888,6 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       25.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
@@ -2946,28 +2925,24 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       5.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
   TipViaCode(
       "site1.com",
       10.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
   TipViaCode(
       "site2.com",
       10.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
   TipViaCode(
       "site3.com",
       10.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
diff --git a/components/brave_rewards/browser/rewards_service_impl.cc b/components/brave_rewards/browser/rewards_service_impl.cc
index efc5947f1..57915102e 100644
--- a/components/brave_rewards/browser/rewards_service_impl.cc
+++ b/components/brave_rewards/browser/rewards_service_impl.cc
@@ -2513,31 +2513,6 @@ void RewardsServiceImpl::GetRewardsInternalsInfo(
                      AsWeakPtr(), std::move(callback)));
 }
 
-void RewardsServiceImpl::OnTip(
-    const std::string& publisher_key,
-    const double amount,
-    const bool recurring,
-    std::unique_ptr<brave_rewards::ContentSite> site) {
-  if (!site) {
-    return;
-  }
-
-  auto info = ledger::PublisherInfo::New();
-  info->id = publisher_key;
-  info->name = site->name;
-  info->url = site->url;
-  info->provider = site->provider;
-  info->favicon_url = site->favicon_url;
-
-  bat_ledger_->SavePublisherInfo(
-      std::move(info),
-      base::BindOnce(&RewardsServiceImpl::OnTipPublisherSaved,
-          AsWeakPtr(),
-          publisher_key,
-          amount,
-          recurring));
-}
-
 void RewardsServiceImpl::OnTipPublisherSaved(
     const std::string& publisher_key,
     const double amount,
diff --git a/components/brave_rewards/browser/rewards_service_impl.h b/components/brave_rewards/browser/rewards_service_impl.h
index f152aabe8..c877fa28c 100644
--- a/components/brave_rewards/browser/rewards_service_impl.h
+++ b/components/brave_rewards/browser/rewards_service_impl.h
@@ -247,12 +247,6 @@ class RewardsServiceImpl : public RewardsService,
 
   void RemoveAllPendingContributionsUI() override;
 
-  void OnTip(
-      const std::string& publisher_key,
-      const double amount,
-      const bool recurring,
-      std::unique_ptr<brave_rewards::ContentSite> site) override;
-
   void OnTip(
       const std::string& publisher_key,
       const double amount,

```

Reply:
hi, sorry forgot about browser tests. Thanks for trying, let's keep it as is. Sorry for a noice

Reply:
NP :)